### PR TITLE
release: v0.1.0-preview.3

### PR DESCRIPTION
**Pull request by Geal**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,113 @@ All notable changes to Router will be documented in this file.
...
+
+- **Trim the query to better detect an empty query** ([PR #738](https://github.com/apollographql/router/pull/738))
+
+  Before this fix, if you wrote a query with only whitespaces inside, it wasn't detected as an empty query.
+
+- **Keep the original context in `RouterResponse` when returning an error** ([PR #738](https://github.com/apollographql/router/pull/738))
+
+  This fix keeps the original http request in `RouterResponse` when there is an error.
+
+- **add a user-agent header to the studio usage ingress submission ([PR #773](https://github.com/apollographql/router/pull/773))
```
Comment (garypen) [inline CHANGELOG.md:83]: Missing the final pair of asterisks.
```suggestion
- **add a user-agent header to the studio usage ingress submission** ([PR #773](https://github.com/apollographql/router/pull/773))
```

File: NEXT_CHANGELOG.md [hunk 2]
```diff
@@ -97,6 +108,9 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
   This fix keeps the original http request in `RouterResponse` when there is an error.
 
+- **add a user-agent header to the studio usage ingress submission ([PR #773](https://github.com/apollographql/router/pull/773))
```
Comment (garypen) [inline NEXT_CHANGELOG.md:111]: Same as above.

**Review by garypen — CHANGES_REQUESTED** [review]
Looks mainly good. I just realised that I never put a fix entry in the CHANGELOG.md for the fix to the benchmarks which have been broken for at least a few weeks. Is it worth adding? It doesn't seem like anyone noticed.

File: NEXT_CHANGELOG.md (hunk 2 above)
Comment (Geal) [reply NEXT_CHANGELOG.md:111]: thanks

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,113 @@ All notable changes to Router will be documented in this file.
...
+
+  You can now provider a `--version or -V` flag to the router. It will output version information and terminate.
+
+- **New startup message** ([PR #780](https://github.com/apollographql/router/pull/780))
+
+  The router startup message was updated with more links to documentation and version information.
+
+- **Add better support of introspection queries** ([PR #802](https://github.com/apollographql/router/pull/802))
+
+  Before this feature the Router didn't execute all the introspection queries, only a small  of the most used ones was executed. Now it detects if it's an introspection query, try to fetch it from cache, if it's not in the cache we execute it and put the response in the cache.
```
Comment (BrynCooke) [inline CHANGELOG.md:19]:
```suggestion
  Before this feature the Router didn't execute all the introspection queries, only a small number of the most common ones were executed. Now it detects if it's an introspection query, try to fetch it from cache, if it's not in the cache we execute it and put the response in the cache.
```

**Comment by Geal** [comment]
I added the benchmarks fix to the changelog in 58fcbd23

File: NEXT_CHANGELOG.md [hunk 4]
```diff
@@ -32,13 +32,16 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
 
   You can now provider a `--version or -V` flag to the router. It will output version information and terminate.
 
+- **New startup message** ([PR #780](https://github.com/apollographql/router/pull/780))
+
+  The router startup message was updated with more links to documentation and version information.
+
 - **Add better support of introspection queries** ([PR #802](https://github.com/apollographql/router/pull/802))
 
   Before this feature the Router didn't execute all the introspection queries, only a small  of the most used ones was executed. Now it detects if it's an introspection query, try to fetch it from cache, if it's not in the cache we execute it and put the response in the cache.
```
Comment (BrynCooke) [inline NEXT_CHANGELOG.md:47]:
```suggestion
  Before this feature the Router didn't execute all the introspection queries, only a small number of the most common ones were executed. Now it detects if it's an introspection query, try to fetch it from cache, if it's not in the cache we execute it and put the response in the cache.
```

**Review by BrynCooke — APPROVED** [review]

**Review by garypen — APPROVED** [review]

**Pull request merged**